src/V2/ApplicationV2/Services/ReadCommandService.cs

```diff
@@ -41,10 +55,16 @@ public Task<ReadValueResponseCollection> ReadValuesAsync(IOpcUaSession opcSessio
             cancellationToken.ThrowIfCancellationRequested();
 
             var response = new ReadValueResponseCollection();
-            //TODO: Talk about error handling -> NULL NodeId            
+            
             var commandsWithRegisteredNodeIds = GetCommandsWithRegisterdNodeIds(opcSession, commands);
             var nodeIds = commands.Select(x => x.NodeId).ToList();
-            var values = opcSession.ReadNodeValues(nodeIds, 10, out var errors);
+
+            var values = new List<object>();
+            var errors = new List<ServiceResult>();
+
+            var batchHandler = new BatchHandler<NodeId>(opcUaConfiguration.ReadBatchSize, ReadValuesInBatch(opcSession, values, errors));
+            batchHandler.RunBatches(nodeIds.ToList());
+            logger.LogTrace("Executed {nrOfNodes} read commands in {nrOfBatches} batch(es).", nodeIds.Count(), (nodeIds.Count + opcUaConfiguration.ReadBatchSize - 1) / opcUaConfiguration.ReadBatchSize);
```

Comment:
We should revisit this as this will have a performance impact regardless if logging is enabled or not.

Reply:
Are you referring to this single occurrence or referring to logging in general?
Is there another way we can build logging that won't have an impact when disabled?
Or are you perhaps referring to the calculation of the number of batches?

Reply:
A bit of both. But in this case we are doing calculations for logging purposes but we do not know yet if the logging will happen (due to log level config). So the calculation will happen regardless of log config.
In this case the effect might be small. But I added a TODO so we are always mindful of the effect such logging can have.

Reply:
Don't think we need to change it now. We should just revisit our logging strategy at some point

Reply:
Thanks for the clarification.
I think regarding the calculation....it is something we can definitely remove.
I think unit tests should have the responsibility to check that the batching is done properly.
We could perhaps just log the total number of items and the batch size. Then you can make the quick calculation from that if you really need to know how many batches executed.

Reply:
Done. Removed the calculation for now and changed the logging slightly.